docs/testplan/transceiver_onboarding/cmis_cdb_firmware_upgrade_testplan.md

```diff
@@ -0,0 +1,445 @@
+   - **Copy only the selected firmware binaries** to the target directory structure on the DUT.
+   - **Validate firmware binary integrity** using MD5 checksums after copying.
+   
+   Fail the test if the specified firmware version doesn't exist.
+
+**Cleanup:**
+
+- The firmware binary folder on the DUT (`/tmp/cmis_cdb_firmware/`) will be deleted after the test module run is complete to ensure a clean state for subsequent tests
+- Cleanup includes removing both the directory structure and any temporary files created during the process
+
+##### 1.1.3 CMIS CDB Firmware Upgrade Tests
+
+**Prerequisites:**
+
+1. **DOM polling must be disabled** to prevent race conditions between I2C transactions and the CDB mode for modules that cannot support CDB background mode.
+2. **Platform-specific processes:** On some platforms, `thermalctld` or similar user processes that perform I2C transactions with the module may need to be stopped during firmware operations.
+3. **Firmware requirements:**
+   - The firmware version specified  by `firmware_versions` test attribute must be available.
+   - All firmware versions must support the CDB protocol for proper testing.
+4. **Module capabilities:** The module must support dual banks for firmware upgrade operations.
+5. **Network connectivity:** The DUT must have network access to the firmware server specified in `cmis_cdb_firmware_base_url.csv` for downloading firmware binaries.
+
+| TC No. | Test | Steps | Expected Results |
+|------|------|------|------------------|
+| 1 | Firmware download validation | 1. Download the next firmware specified in `firmware_versions` using the sfputil CLI<br>2. Wait until CLI execution completes | 1. CLI execution should finish within `transceiver_download_timeout_minutes` minutes and return 0 <br>2. Active FW version should remain unchanged<br>3. Inactive FW version should reflect the downloaded firmware version<br> 4. No link flap should be seen<br>5. The kernel has no error messages in syslog<br>6. Critical process such as `xcvrd`, `syncd`  `orchagent` does not crash/restart. |
+| 2 | Firmware activation validation | 1. Shut down all the interfaces part of the physical ports<br>2. Execute firmware run<br>3. Execute firmware commit<br>4. Reset the transceiver and wait for 5 seconds<br>5. Startup all the interfaces in Step 1 | 1. The return code on step 2 and 3 is 0 (Return code 0 indicates success)<br>2. Active firmware version should now match the previous inactive firmware version<br>3. Inactive firmware version should now match the previous active firmware version<br>4. Link should be up within `port_wait_time_after_startup_sec` seconds.<br>5. `sfputil show fwversion` CLI now should show the “Committed Image” to the current active bank<br>6. Critical process such as `xcvrd`, `syncd`  `orchagent` does not crash/restart. |
```

Comment:
@az-pz Can we verify the active and inactive FW image bank after each FW run and commit to ensure it is updated inline with expectation?

Reply:
Split this test into two and added verifications steps for image banks.